sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/ChatRequestSystemMessage.java

```diff
@@ -111,7 +118,15 @@ public static ChatRequestSystemMessage fromJson(JsonReader jsonReader) throws IO
                 String fieldName = reader.getFieldName();
                 reader.nextToken();
                 if ("content".equals(fieldName)) {
-                    content = reader.getNullable(nonNullReader -> BinaryData.fromObject(nonNullReader.readUntyped()));
+                    if (reader.currentToken() == JsonToken.STRING) {
+                        content = BinaryData.fromString(reader.getString());
+                    } else if (reader.currentToken() == JsonToken.START_ARRAY) {
+                        content = BinaryData.fromObject(
+                            reader.readArray(arrayReader -> arrayReader.readObject(ChatMessageContentItem::fromJson)));
+                    } else {
+                        throw new IllegalStateException("Unexpected 'content' type found when deserializing"
+                            + " ChatRequestSystemMessage JSON object: " + reader.currentToken());
+                    }
```

Comment:
Similarly as for `ChatRequestAssistantMessage.java` we may want to be less strict here.